gammapy/background/energy_offset_array.py

```diff
@@ -0,0 +1,135 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import numpy as np
+from astropy.units import Quantity, UnitsError
+from astropy.coordinates import Angle
+from astropy.table import Table
+from astropy.units import Quantity
+from astropy.coordinates import Angle
+
+__all__ = [
+    'EnergyOffsetArray',
+]
+
+
+class EnergyOffsetArray(object):
+    """
+    Energy offset dependent array.
+
+    Parameters
+    ----------
+    energy : `~astropy.units.Quantity`
+         energy bin vector
+    offset : `~astropy.coordinates.Angle`
+        offset bin vector
+    data : `~numpy.ndarray`
+        data array (2D)
+
+    """
+
+    def __init__(self, energy=None, offset=None, data=None):
```

Comment:
I think the default `None` value for `energy` and `offset` should be removed.
These are required arguments, i.e. not giving something will result in an error on the next lines, i.e. is not useful, no?

`data` could remain optional, and if no data array is given, a numpy float64 array of the correct shape is constructed (use `len(energy)` and `len(offset)` to compute the shape.